napari/layers/shapes/_shapes_models/shape.py

```diff
@@ -423,11 +423,7 @@ def to_mask(self, mask_shape=None, zoom_factor=1, offset=[0, 0]):
                         self.slice_key[0, j], self.slice_key[1, j] + 1
                     )
                 j += 1
-            # equivalent to: displayed_order = np.argsort(self.dims_displayed)
-            dims_displayed = self.dims_displayed
-            displayed_order = sorted(
-                range(len(dims_displayed)), key=lambda x: dims_displayed[x]
-            )
+            displayed_order = np.argsort(self.dims_displayed)
```

Comment:
Same as above (`argsort` vs `sorted(range...`)

Reply:
I turned our simple `argsort` in `utils.misc` into a public function, documented it, and used it here to avoid a performance regression.